libsolidity/ast/Types.cpp

```diff
@@ -138,9 +139,9 @@ void Type::clearCache() const
 	m_stackSize.reset();
 }
 
-void StorageOffsets::computeOffsets(TypePointers const& _types)
+void StorageOffsets::computeOffsets(TypePointers const& _types, u256 _slotStart)
 {
-	bigint slotOffset = 0;
+	bigint slotOffset = bigint(_slotStart);
```

Comment:
This breaks `storageSize()`, because `m_storageSize` now contains more than just size of the object itself.

Something should be failing because of this. Are we missing a semantic test?

Reply:
I separated the `slotOffset` from the actual `storageSize` and now it should be the correct size.
This also made me realize that we missed checking for cases where the contract is not too large for storage (max capacity) but do not fit due to the specified base slot value.
I added a commit fixing that, maybe small enough to be kept within this PR, but otherwise I can just create a brand new one.

Reply:
Yes, it should definitely be a part of this PR.